src/filecoin/FilecoinEngine.ts

```diff
@@ -312,59 +339,62 @@ export class FilecoinEngine extends CurrencyEngine<
         this.onUpdateBlockHeight(tx.blockHeight)
       }
 
-      this.tokenCheckTransactionsStatus[this.currencyInfo.currencyCode] =
-        progress / scanners.length
-      this.updateOnAddressesChecked()
+      // Handle progress update the progress
+      progressMap[scanId] = progress
+      handleScanProgress()
     }
 
     const scanners = [
-      this.scanTransactionsFromFilscan(addressString, handleScan)
+      this.scanTransactions(
```

Comment:
This entire commit is overkill. Just create a `scanTransactionsFromFilfox` and add that to the `scanners` array. Set the progress = 1 once all scanners are done.

Reply:
There is more code dup, but that works just fine. I kept the progress during scan, but instead of figuring out what the scan progress is combined by all scanners, I just took the greater progress of the two scanners as progress.